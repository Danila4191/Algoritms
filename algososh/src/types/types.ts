export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}
export enum Direction {
  Ascending = "ascending",
  Descending = "descending",
}
export interface arrString {
  value: string;
  state: ElementStates;
  head?: boolean,
  tail?: boolean,
  smalCircle?:smalCircle
}
export interface smalCircle {
  value: string;
  state: ElementStates;
  position: string
}
export interface arrSortingnumbers {
  value: number;
  state: ElementStates;
}

export interface IStack<T> {
  push: (item: arrString) => void;
  pop: () => void;
  clear: () => void;
  getSize: () => number;
}
export interface IQueue<T> {
  enqueue: (item: arrString) => void;
  dequeue: () => void;
}
export  interface ILinkedList<T> {
  append: (item: any) => void;
  prepend: (item:any) => void;
  addByIndex: (item: any, position: number) => void;
  deleteByIndex: (index: number) => void;
  find: (index: number) => void;
  getSize: () => number;
}