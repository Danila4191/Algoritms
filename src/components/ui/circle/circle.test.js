import renderer from "react-test-renderer";
import  {render, screen, fireEvent}  from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/types"; 
it("рендер круга без символов",() => {
    const tree = renderer
    .create(   <Circle   
        letter=""
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it("рендер круга с символом",() => {
    const tree = renderer
    .create(   <Circle   
        letter="123"
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it("рендер круга с Хед",() => {
    const tree = renderer
    .create(   <Circle   
        head="321"
        letter="123"
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it("рендер круга с компонентом в Хед",() => {
    const tree = renderer
    .create(   <Circle   
        head={ <Circle   
            head="smallHead"
            letter="small"
          />}
        letter="123"
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it("рендер круга с Таил",() => {
    const tree = renderer
    .create(   <Circle   
        tail="321"
        letter="123"
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it("рендер круга с компонентом в Таил",() => {
    const tree = renderer
    .create(   <Circle   
        tail={ <Circle   
            tail="smallTail"
            letter="small"
          />}
        letter="123"
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it("рендер круга с индексом",() => {
    const tree = renderer
    .create(   <Circle   
        index="0"
        letter="123"
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it("рендер круга small",() => {
    const tree = renderer
    .create(   <Circle   
     
        isSmall={true}
        letter="123"
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it("рендер круга default",() => {
    const tree = renderer
    .create(   <Circle   
        state={ElementStates .Default}
        letter="123"
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it("рендер круга changing",() => {
    const tree = renderer
    .create(   <Circle   
        state={ElementStates.Changing}
        letter="123"
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
  it("рендер круга modified",() => {
    const tree = renderer
    .create(   <Circle   
        state={ElementStates.Modified}
        letter="123"
      />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })

