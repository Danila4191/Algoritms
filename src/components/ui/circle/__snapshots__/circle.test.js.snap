// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`рендер круга changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`рендер круга default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`рендер круга modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`рендер круга small 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`рендер круга без символов 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`рендер круга с Таил 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    321
  </div>
</div>
`;

exports[`рендер круга с Хед 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    321
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`рендер круга с индексом 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`рендер круга с компонентом в Таил 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          small
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 string"
      >
        smallTail
      </div>
    </div>
  </div>
</div>
`;

exports[`рендер круга с компонентом в Хед 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head string"
      >
        smallHead
      </div>
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          small
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`рендер круга с символом 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
